quinn-proto/src/endpoint.rs

```diff
@@ -466,8 +467,41 @@ impl Endpoint {
             )));
         }
 
-        let (retry_src_cid, orig_dst_cid) = if server_config.use_retry {
-            if token.is_empty() {
+        let (retry_src_cid, orig_dst_cid) = 'accept: {
```

Comment:
At about 7-8 levels of indentation, maybe it's time to split this logic out?

Reply:
Split the `match` out, which simplified the control flow considerably. I didn't split the retry composition out because it needs a lot of locals.